### PR TITLE
extend YearMonth

### PR DESCRIPTION
**Pull request by 9il**

### CODE REVIEW

File: source/mir/date.d [hunk 1]
```diff
@@ -572,35 +574,37 @@ struct YearMonth
...
-    version (mir_test)
-    @safe unittest
-    {
-        auto ym = YearMonth(2000, Month.dec);
-    }
-
-    ///
-    this(short year, Quarter quarter, AssumePeriod assumePeriod = AssumePeriod.begin) @safe pure nothrow @nogc
+    @reflectIgnore!"SIL" // clashes with ushort version
+    this(short year, Month month) @safe pure nothrow @nogc
```
Comment (jmh530) [inline source/mir/date.d:591]: @9il This might have been a typo on my part. I think there was mean to be a version that takes a `Quarter` as an input.

**Comment by 9il** [comment]
The YearMonth constructor overload with Quarter has been removed because of conflict with another constructor

**Comment by jmh530** [comment]
It might also be good if `YearMonthDay` had an `endOfMonth` member function, corresponding to the `YearMonth`/`Date` member functions. There is a bit of an issue though in that both of those return a `Date`, but what if you want to return a `YearMonthDay`. I'm tempted to say that there should be two functions, one to return a `Date` and one to return a `YearMonthDay`.

**Comment by 9il** [comment]
YearMonthDay is just an intermediate representation for some algorithms.

**Comment by jmh530** [comment]
So long as it is public, people would (or could) use it...

In the back of my mind, I am just thinking about how useful I find the `EOMONTH` function in Excel, which basically chains `add!"month"` and `endOfMonth` together.

**Pull request merged**